### ISSUE TITLE
WebClient has odd behavior with the host header.

### ISSUE THREAD

**Issue by lscoughlin** [oracle/helidon#1668]
## Environment Details
* Helidon Version: 2.0.0-M2
* Helidon SE or Helidon MP
* JDK version: 11.0.6
* OS: Mac OS X 10.14.6
* Docker version (if applicable): N/A

----------
If you set the host header like this:

```
            webclient.method("GET")
                    .headers((h) -> {
                        h.put(HttpHeaderNames.HOST.toString(), "some.virtual.host");
                        return h;
                    })
```

it spits out two host headers like this:
host                          : localhost
host                          : some.virtual.host

If you set the host header like this:

```
            webclient.method("GET")
                    .headers((h) -> {
                        h.put("Host", "some.virtual.host");
                        return h;
                    })
```
it spits out only 1 ( correct ) host header.

**Comment by lscoughlin** [comment]
Gist showing the issue [here](https://gist.github.com/lscoughlin/abbcb4707fb20dab44e24a452cfffba0):

**Comment by Verdent** [comment]
Hi, thank you this find :-) Actually I made fix for that and it is currently under review with some of other changes. It should be fixed in the new release.

**Comment by Verdent** [comment]
Fixed in master

**Comment by romain-grecourt** [comment]
This will be available in the next version (2.0.0-M3)
